### ISSUE TITLE
Disable installing of Lighttpd through an env variable

### ISSUE THREAD

**Issue by TheAntimist** [pi-hole/pi-hole#2349]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:** 

`4`

---

This is a feature request. I use nginx, instead of lighttpd, thus, it would help if I can have a way of disabling the install of lighttpd, because now, I need to perform a package purge every time I update. (Although, I have scripted this for updating)

Thus, my request is that, can we introduce an environment variable, say, `DISABLE_LIGHTTPD` which when set, should not check for lighttpd and install it, along with all the lighttpd specific settings.

I know this might be a bit of work, but I feel that this will help all the users who do not use Lighttpd, by default.

**Comment by rrobgill** [comment]
Before writing something new, it might be worth your while to have a look at:

https://github.com/pi-hole/pi-hole/blob/ddbdb51d209fa9eea2451a1bebfaeaef23a9b77b/automated%20install/basic-install.sh#L80

There may be something there that suits your needs. Not environment variable based, but command line options are pretty workable too.

**Comment by TheAntimist** [comment]
Thanks for pointing that out, I actually did look at it but had doubts on whether `INSTALL_WEB_SERVER` would affect `INSTALL_WEB_INTERFACE`. But looking at it again, I think it might just work.

**Comment by sangaman** [comment]
It would be nice if the `INSTALL_WEB_SERVER`flag were documented. I'll try it next time I update as it is a hassle to have to uninstall lighttpd every time I update (as I am also using nginx).

**Comment by Mcat12** [comment]
It is automatically set by the installer when you choose to not install the web server, it is not user-facing.

**Comment by sangaman** [comment]
> It is automatically set by the installer when you choose to not install the web server (there is a prompt), it is not user-facing.

Is this just for first time installs? Just now running `pihole -up` it installed lighttpd without prompting me.

**Comment by PromoFaux** [comment]
Yeah, you are only prompted on first time install (or if you reconfigure using `pihole -r`.

You can also set it manually in `/etc/pihole/setupvars.conf`
